File: packages/types/src/events.ts
export type ThinkingEvent = {
  type: 'thinking';
  status: string;
};

export type PersonasDiscoveredEvent = {
  type: 'personas_discovered';
  personas: Array<{
    persona_id?: string;
    name: string;
    segment: string;
    confidence: number;
  }>;
};

export type ContentDeltaEvent = {
  type: 'content_delta';
  delta: string;
  persona_id: string;
};

export type SourcesEvent = {
  type: 'sources';
  persona_id: string;
  sources: Array<{
    chunk_id: string;
    document_id: string;
    title: string;
    confidence: number;
    excerpt: string;
  }>;
};

export type CompleteEvent = {
  type: 'complete';
  persona_id: string;
  latency_ms: number;
};

export type PersonaSwitchEvent = {
  type: 'persona_switch';
  persona_id: string;
  name: string;
};

export type ChatEvent =
  | ThinkingEvent
  | PersonasDiscoveredEvent
  | ContentDeltaEvent
  | SourcesEvent
  | CompleteEvent
  | PersonaSwitchEvent;

File: packages/proto/src/msqdx_glass_proto/events.py
from __future__ import annotations

from typing import List

from pydantic import BaseModel, Field


class ThinkingEvent(BaseModel):
    type: str = Field("thinking", frozen=True)
    status: str


class PersonaSummary(BaseModel):
    persona_id: str | None = None
    name: str
    segment: str
    confidence: float


class PersonasDiscoveredEvent(BaseModel):
    type: str = Field("personas_discovered", frozen=True)
    personas: List[PersonaSummary]


class ContentDeltaEvent(BaseModel):
    type: str = Field("content_delta", frozen=True)
    delta: str
    persona_id: str


class SourceEntry(BaseModel):
    chunk_id: str
    document_id: str
    title: str
    confidence: float
    excerpt: str


class SourcesEvent(BaseModel):
    type: str = Field("sources", frozen=True)
    persona_id: str
    sources: List[SourceEntry]


class CompleteEvent(BaseModel):
    type: str = Field("complete", frozen=True)
    persona_id: str
    latency_ms: int


class PersonaSwitchEvent(BaseModel):
    type: str = Field("persona_switch", frozen=True)
    persona_id: str
    name: str


ChatEvent = (
    ThinkingEvent
    | PersonasDiscoveredEvent
    | ContentDeltaEvent
    | SourcesEvent
    | CompleteEvent
    | PersonaSwitchEvent
)

